llvm/include/llvm/CodeGen/GlobalISel/CombinerHelper.h

```diff
@@ -256,8 +257,11 @@ class CombinerHelper {
   /// concat_vectors.
   ///
   /// \pre MI.getOpcode() == G_SHUFFLE_VECTOR.
-  bool matchCombineShuffleVector(MachineInstr &MI,
-                                 SmallVectorImpl<Register> &Ops);
+  using GeneratorType = std::function<std::optional<int32_t>()>;
+
+  bool matchCombineShuffleVector(MachineInstr &MI, GeneratorType Generator,
```

Comment:
The docstring for `matchCombineShuffleVector` is now physically disconnected from the function. Can you fix it? I think some documentation for `GeneratorType` would also be helpful, it's not immediately clear to me what it represents.

Reply:
Updated the documentation and added some for GeneratorType. Thanks for pointing it out :)